enterprise/frontend/src/embedding-sdk/components/public/InteractiveDashboard/InteractiveDashboard.tsx

```diff
@@ -110,6 +112,16 @@ const InteractiveDashboardInner = ({
   );
 };
 
-export const InteractiveDashboard = withPublicComponentWrapper(
-  InteractiveDashboardInner,
-);
+export const InteractiveDashboard =
+  withPublicComponentWrapper<InteractiveDashboardProps>(
+    ({ dashboardId, ...rest }) => {
+      const id = useValidatedEntityId({
+        type: "dashboard",
```

Comment:
There is one DX improvement we could do here (I'm not sure how complicated this would be tho), since we know the type and possible values are finite. We could probably infer the return type automatically.

Imagine passing `type: "dashboard"` and the hook returning `DashboardId`, that'd be AWESOMEEE.

One pro if we pulled that out is to avoid human error since the caller could practically cast the return value to any type.

Reply:
I took like 3-4 hours trying to figure that one out and I can't seem to get it to work 🙃  basically, after creating the mapped types + generics, the TS compiler would just return `any`. I'm really sad about it

Reply:
No worries then. I don't think it'd be that simple 😓